adapters/Snap/SnapAdapter/GADMediationAdapterSnap.m

```diff
@@ -14,71 +14,102 @@
-  completionHandler("", nil);
+@implementation GADMediationAdapterSnap {
+    GADSAKAdViewContainer *_banner;
+    GADSAKInterstitialContainer *_interstitial;
+    GADSAKRewardedAdContainer *_rewardedAd;
 }
 
-+ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
-             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Snap ad network SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
++ (GADVersionNumber)adapterVersion
+{
+    NSArray<NSString *> *versionComponents = [GADMAdapterSnapVersion componentsSeparatedByString:@"."];
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 3) {
+      version.majorVersion = versionComponents[0].integerValue;
+      version.minorVersion = versionComponents[1].integerValue;
+      version.patchVersion = versionComponents[2].integerValue;
+    }
+    return version;
+}
+
++ (GADVersionNumber)adSDKVersion
+{
+    NSArray<NSString *> *versionComponents = [SAKMobileAd.shared.sdkVersion componentsSeparatedByString:@"."];
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 3) {
+      version.majorVersion = versionComponents[0].integerValue;
+      version.minorVersion = versionComponents[1].integerValue;
+      version.patchVersion = versionComponents[2].integerValue;
+    }
+    return version;
 }
 
-+ (GADVersionNumber)adSDKVersion {
-  // TODO: Populate `versionString` with the ad-network's SDK version in NSString format.
-  NSString *versionString = @"";
-  GADVersionNumber version = {0};
-  NSArray<NSString *> *components = [versionString componentsSeparatedByString:@"."];
-  if (components.count == 3) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue;
-  } else {
-    NSLog(@"Unexpected ad SDK version string: %@. Returning 0 for adSDKVersion.", versionString);
-  }
-  return version;
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass
+{
+    return nil;
 }
 
-+ (GADVersionNumber)adapterVersion {
-  GADVersionNumber version = {0};
-  NSArray<NSString *> *components = [GADMAdapterSnapVersion componentsSeparatedByString:@"."];
-  if (components.count == 4) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue * 100 + components[3].integerValue;
-  }
-  return version;
++ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
+             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler
+{
+    for (GADMediationCredentials *credentials in configuration.credentials) {
+        NSString *snapAppId = credentials.settings[GADMAdapterSnapAppID];
+        if (snapAppId.length) {
```

Comment:
We should probably add error handling when multiple app Ids were found.

Reply:
How can this be possible?

Reply:
It is possible considering this scenario: Under the same app in AdMob account, the publisher might (accidentally) use different Snap App IDs for two different AdMob ad units.

Implementation example: https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Fyber/FyberAdapter/GADMediationAdapterFyber.m#L61-L85

Reply:
Thanks for the explanation! Addressed

Reply:
Thank you!